### PR TITLE
fix: running full formatting checks

### PR DESCRIPTION
**Pull request by gentlementlegen**
The full formatting checks were never run during PRs.

### CODE REVIEW

File: .github/workflows/formatting-checks.yml [hunk 1]
```diff
@@ -17,8 +17,7 @@ jobs:
...
 
-      - name: Install cspell
-        run: yarn add cspell
-
-      - name: Run cspell
-        run: yarn format:cspell
+      - name: Run formatting checks
+        run: |
+          yarn install
+          yarn format
```
Comment (0x4007) [inline .github/workflows/formatting-checks.yml]: I think a better idea is to run checker only which is faster. Something like `eslint --check` etc.

This format script actually writes changes to the source code as it is intended to be run developer side before committing.

Comment (gentlementlegen) [reply .github/workflows/formatting-checks.yml]: We need to run at least cspell, eslint, probably tsc as well to make sure it compiles. True the code is changed but nothing get committed and you can see the affected files. I'll have a look.

Comment (gentlementlegen) [reply .github/workflows/formatting-checks.yml]: @0x4007 How about https://github.com/gentlementlegen/plugin-template/actions/runs/11233078532/job/31226031158

Comment (0x4007) [reply .github/workflows/formatting-checks.yml]: I would make a script in package.json like format:check

run-p cspell:check eslint:check prettier:check 

And then fill those sub scripts out accordingly.

File: eslint.config.mjs [hunk 2]
```diff
@@ -65,6 +69,12 @@ export default tsEslint.config({
     "sonarjs/no-element-overwrite": "error",
     "sonarjs/no-identical-conditions": "error",
     "sonarjs/no-identical-expressions": "error",
+    "sonarjs/new-cap": "off",
+    "sonarjs/different-types-comparison": "off",
+    "sonarjs/sonar-prefer-regexp-exec": "off",
+    "sonarjs/function-return-type": "off",
+    "sonarjs/no-misleading-array-reverse": "off",
+    "sonarjs/slow-regex": "off",
```
Comment (0x4007) [inline eslint.config.mjs:77]: All new rules?

**Review by 0x4007 — APPROVED** [review]

File: eslint.config.mjs (hunk 2 above)
Comment (gentlementlegen) [reply eslint.config.mjs:77]: Due to `sonar` update they introduced news rules that were annoying (in my opinion) so I disabled them.

Comment (0x4007) [reply eslint.config.mjs:77]: Something feels off about having to manually disable rules. I understand normally it's an opt in list. I don't understand why new rules would have been by default applied. 

Anyways I looked these up and I'm fine to disable them.

Comment (gentlementlegen) [reply eslint.config.mjs:77]: Some of them felt bad, like `function-return-type` not letting you return `A | B`, or slow regex that would literally forbid any regex that use forward lookup etc. Most of these were warnings not errors but they cluttered the output in my opinion.

**Pull request merged**